src/concrete_convex_hull.jl

```diff
@@ -99,41 +99,64 @@ function convex_hull!(points::Vector{VN};
 
     n = length(first(points))
 
-    # two points
-    if m == 2
-        if n == 1
-            p1, p2 = points[1], points[2]
-            if p1 == p2  # check for redundancy
-                pop!(points)
-            elseif p1[1] > p2[1]
-                points[1], points[2] = p2, p1
-            end
-        elseif n == 2
-            # special case, see #876
-            p1, p2 = points[1], points[2]
-            if p1 == p2  # check for redundancy
-                pop!(points)
-            elseif p1 <= p2
-                nothing
-            else
-                points[1], points[2] = p2, p1
-            end
+    if n == 1 # dimensional check
+        if m == 2
+            return _two_points_1d!(points) # two points case in 1d
+        else
+            return _convex_hull_1d!(points) # general case in 1d
         end
-        return points
+    elseif n == 2
+        if m == 2 # two points case in 2d
+            return _two_points_2d!(points)
+        elseif m == 3 # three points case in 2d
+            return _three_points_2d!(points)
+        else # general case in 2d
+            return _convex_hull_2d!(points, algorithm=algorithm)
+        end
+    else # general case in nd
+        return _convex_hull_nd!(points, backend=backend)
     end
+end
 
-    # ===========================
-    # Dispatch for general cases
-    # ===========================
+function _two_points_1d!(points)
+    p1, p2 = points[1], points[2]
+    if p1 == p2  # check for redundancy
+        pop!(points)
+    elseif p1[1] > p2[1]
+        points[1], points[2] = p2, p1
+    end
+    return points
+end
+
+function _two_points_2d!(points)
```

Comment:
I think the implementation of `_two_points_2d!` also works in the 1D case.
```julia
julia> [2.] <= [2.]
true

julia> [2.1] <= [2.]
false
```
So I think you can remove the 1D case and call the same function.

Reply:
LazySets overrides the `<=` [fallback for vectors](https://github.com/JuliaLang/julia/blob/master/base/abstractarray.jl#L1759) so it doesn't work:
 
```julia
julia> [2.] <= [2.]
true

julia> using LazySets

julia> [2.] <= [2.]
ERROR: BoundsError: attempt to access 1-element Array{Float64,1} at index [2]
Stacktrace:
 [1] getindex at ./array.jl:729 [inlined]
 [2] <=(::Array{Float64,1}, ::Array{Float64,1}) at /Users/forets/.julia/dev/LazySets/src/AbstractPolygon.jl:199
 [3] top-level scope at none:0
```

I would suggest to keep this code split (another approach would be to use a branch in the `<=` code, but i don't support this idea at least not before checking if it can be done without regression).